app/tests/test_browse_consignment.py

```diff
@@ -0,0 +1,340 @@
+                        "'10/03/2023', 'third_file.docx', 'Closed', '10/03/2023'"
+                    ],
+                ],
+            ),
+            (
+                "sort=closure_type-desc&record_status=all&date_filter_field=date_last_modified"
+                "&date_from_day=01&date_from_month=01&date_from_year=2020",
+                [
+                    [
+                        "'20/05/2023', 'fifth_file.doc', 'Open', '-', "
+                        "'15/01/2023', 'second_file.ppt', 'Open', '-', "
+                        "'25/02/2023', 'first_file.docx', 'Closed', '25/02/2023', "
+                        "'12/04/2023', 'fourth_file.xls', 'Closed', '12/04/2023', "
+                        "'10/03/2023', 'third_file.docx', 'Closed', '10/03/2023'"
+                    ],
+                ],
+            ),
+            (
+                "sort=closure_type-desc&record_status=all&date_filter_field=date_last_modified"
+                "&date_to_day=10&date_to_month=03&date_to_year=2023",
+                [
+                    [
+                        "'15/01/2023', 'second_file.ppt', 'Open', '-', "
+                        "'25/02/2023', 'first_file.docx', 'Closed', '25/02/2023', "
+                        "'10/03/2023', 'third_file.docx', 'Closed', '10/03/2023'"
+                    ],
+                ],
+            ),
+            (
+                "sort=closure_type-desc&record_status=all&date_filter_field=opening_date"
+                "&date_from_day=01&date_from_month=01&date_from_year=2023",
+                [
+                    [
+                        "'25/02/2023', 'first_file.docx', 'Closed', '25/02/2023', "
+                        "'12/04/2023', 'fourth_file.xls', 'Closed', '12/04/2023', "
+                        "'10/03/2023', 'third_file.docx', 'Closed', '10/03/2023'"
+                    ],
+                ],
+            ),
+            (
+                "sort=opening_date-asc&record_status=all&date_filter_field=opening_date"
+                "&date_to_day=03&date_to_month=04&date_to_year=2023",
+                [
+                    [
+                        "'25/02/2023', 'first_file.docx', 'Closed', '25/02/2023', "
+                        "'10/03/2023', 'third_file.docx', 'Closed', '10/03/2023'"
+                    ],
+                ],
+            ),
+            (
+                "sort=closure_type-desc&record_status=all&date_filter_field=opening_date"
+                "&date_from_day=10&date_from_month=03&date_from_year=2023&date_to_day=20"
+                "&date_to_month=10&date_to_year=2023",
+                [
+                    [
+                        "'12/04/2023', 'fourth_file.xls', 'Closed', '12/04/2023', "
+                        "'10/03/2023', 'third_file.docx', 'Closed', '10/03/2023'"
+                    ],
+                ],
+            ),
+        ],
+    )
+    def test_browse_consignment_full_test(
```

Comment:
paramterization is helping greatly, makes these tests much easier to follow! Like we discussed, if we get a chance after delivering MVP we may revist to move the testing of edge case of filters and sort combinations to a lower level and leave fewer higher level route tests but already the paramterization is a big help.

Reply:
yes, I agree, we can perform sort and filter combinations on lower level